nx/lib/nx.ex

```diff
@@ -6401,8 +6474,36 @@ defmodule Nx do
         Nx.Shape.broadcast_axes(false_shape, output_shape)
       )
 
-    out = %{pred | shape: output_shape, type: output_type, names: output_names}
-    impl!(pred, on_true, on_false).select(out, pred, on_true, on_false)
+    out = devectorize(%{pred | shape: output_shape, type: output_type, names: output_names})
+
+    pred = devectorize(pred)
```

Comment:
Maybe invert the order of these two operations so you don't have to devectorize `out`?

```suggestion
    pred = devectorize(pred)
    out = %{pred | shape: output_shape, type: output_type, names: output_names}
```

?

Reply:
We'd have to recalculate `output_shape` if we invert the order